opentelemetry-sdk/src/opentelemetry/sdk/instrumentation/__init__.py

```diff
@@ -0,0 +1,167 @@
+# Copyright The OpenTelemetry Authors
```

Comment:
@owais Added a comment here so we can have a thread to discuss

> I think in that case adding to SDK is much more simpler.

Agreed. The `Configurator` depends on the SDK **a lot**

https://github.com/open-telemetry/opentelemetry-python/blob/f11ed2f3bacb11d53a7a2b4837cf6308fa34cc71/opentelemetry-distro/src/opentelemetry/distro/__init__.py#L29-L32

and `opentelemetry-instrumentation` is not allowed to depend on `opentelemetry-sdk` 😕  https://github.com/open-telemetry/opentelemetry-python/issues/1421

> We can also make the whole package private by naming it something like opentelemetry.sdk._configuration. This will buy us time to come up with a better name later.

Totally agree with you, I wasn't too attached with `opentelemetry.sdk.instrumentation` anyways 🙂 

I think a private package is a great idea. Also, the `Configurators` are only ever used in the context of "auto-instrumentation". It's too long to add that in the name, but I think as people are more familiar with AI, it will make sense to them.

I'm worried `._configuration` might also be a little confusing. what if we put it in `opentelemetry.sdk._auto_instrumentation`? There's already a `opentelemetry.instrumentation._auto_instrumentation` but no one uses it as no one should really (to be totally automatic!) 🙂 

Let me know what you think!

Reply:
I'd happy with anything except `instrumentation` in the package name. I think distro/configurator makes auto-instrumentation easier but that's not it's purpose. It's main purpose is the auto-configure the SDK based on defaults and user provided environment variables. Even if a user does not install any instrumentation packages, this is still very useful to configure tracing pipeline automatically.

Reply:
> Even if a user does not install any instrumentation packages, this is still very useful to configure tracing pipeline automatically.

Hm but we're supposed to be discouraging the use of it for now right? Hence why it's private? I guess it's fine because like you said we don't need to commit to maintaining it. Maybe one day it'll move into the main `opentelemetry.sdk` path and make configuration easy in general not just for auto instrumentation like you said.

> I'd happy with anything except instrumentation in the package name.

That's far, I think `._configurator` or `._configuration` are both good. Since you suggested the latter I will change it to that!

Reply:
> Hm but we're supposed to be discouraging the use of it for now right? Hence why it's private? 

I don't think we are discouraging other distros from using it but it serves as a warning that interface may change in any release without any warnings. That covers us and doesn't lock us into what we merge today.